engine.py

```diff
@@ -161,114 +160,72 @@ def log_error(self, msg):
         self._log.error(msg)
 
     ##########################################################################################
-    # pyside / qt
-    
-    def _define_qt_base(self):
-        """
-        check for pyside then pyqt
-        """
+    # qt
+
+    def _define_unavailable_base(self):
         # proxy class used when QT does not exist on the system.
         # this will raise an exception when any QT code tries to use it
-        class QTProxy(object):                        
+        class QTProxy(object):
             def __getattr__(self, name):
                 raise tank.TankError("Looks like you are trying to run an App that uses a QT "
                                      "based UI, however the Shell engine could not find a PyQt "
-                                     "or PySide installation in your python system path. We " 
+                                     "or PySide installation in your python system path. We "
                                      "recommend that you install PySide if you want to "
                                      "run UI applications from the Shell.")
+        return {
```

Comment:
Should this be coming from core so we wouldn't need to update tk-shell in the future if we add modules?

Reply:
Based on conversation [here](https://github.com/shotgunsoftware/tk-shotgun/pull/6#discussion_r181482082) it seems like returning the proxy object from core may not be viable at this point?